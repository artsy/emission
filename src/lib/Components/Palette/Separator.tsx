// @ts-ignore
import React from "react"

import { color } from "@artsy/palette"
import styled from "styled-components/native"
import { space, SpaceProps, width, WidthProps } from "styled-system"

interface SeparatorProps extends SpaceProps, WidthProps {}

export const Separator = styled.View.attrs<SeparatorProps>({})`
  border: 1px solid ${color("black10")};
  border-bottom-width: 0;
  ${space};
  ${width};
`

Separator.defaultProps = {
  width: "100%",
}
