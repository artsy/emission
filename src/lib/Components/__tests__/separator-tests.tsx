import * as React from "react"
import "react-native"
import * as renderer from "react-test-renderer"
import Separator from "../separator"

it("looks like expected", () => {
  const tree = renderer.create(<Separator />).toJSON()
  expect(tree).toMatchSnapshot()
})
