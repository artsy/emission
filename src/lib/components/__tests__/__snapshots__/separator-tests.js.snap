exports[`test looks like expected 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#e5e5e5",
        "height": 1,
        "marginLeft": -20,
        "marginRight": -20,
      },
      undefined,
    ]
  } />
`;
